test/gleam/bit_array_test.gleam

```diff
@@ -308,3 +308,30 @@ pub fn inspect_partial_bytes_test() {
   bit_array.inspect(<<5:3, 11:4, 1:2>>)
   |> should.equal("<<182, 1:size(1)>>")
 }
+
+pub fn find_found_test() {
+  let needle = <<"bar":utf8>>
+  <<"foobar":utf8>>
+  |> bit_array.find(needle)
+  |> should.be_ok
+  |> should.equal(3)
+}
+
+pub fn find_not_found_test() {
+  let needle = <<"baz":utf8>>
+  <<"foobar":utf8>>
+  |> bit_array.find(needle)
+  |> should.be_error
+}
+
+pub fn find_empty_needle_test() {
+  <<"foobar":utf8>>
+  |> bit_array.find(<<>>)
+  |> should.be_error
+}
+
+pub fn find_empty_haystack_test() {
+  <<>>
+  |> bit_array.find(<<"bar":utf8>>)
+  |> should.be_error
+}
```

Comment:
Let's also have some tests for non-byte aligned bit arrays. These tests would need to have the `@target(erlang)` attribute as currently the JavaScript target only supports byte aligned bit arrays.

Reply:
These are still needed! Thank you